content/blog/readr-1-4-0/index.Rmd

```diff
@@ -0,0 +1,109 @@
+---
+output: hugodown::hugo_document
+
+slug: readr-1-4-0
+title: readr 1.4.0
+date: 2020-09-30
+author: Jim Hester
+description: >
+    A 2-3 sentence description of the post that appears on the articles page.
+    This can be omitted if it would just recapitulate the title.
+photo:
+  url: https://unsplash.com/photos/XOW1WqrWNKg
+  author: Anastasia Zhenina
+categories: [package]
+tags:
+  - readr
+  - tidyverse
+---
+
+[readr](http://readr.tidyverse.org) 1.4.0 is now available on CRAN! Learn more
+about readr at <https://readr.tidyverse.org>. Detailed notes are always in the
+[change log](https://readr.tidyverse.org/news/index.html#readr-1-4-0).
+
+The readr package makes it easy to get rectangular data out of comma separated
+(csv), tab separated (tsv) or fixed width files (fwf) and into R. It
+is designed to flexibly parse many types of data found in the wild, while still
+cleanly failing when data unexpectedly changes. If you are new to readr, the
+best place to start is the [data import
+chapter](https://r4ds.had.co.nz/data-import.html) in R for data science.
+
+The easiest way to install the latest version from CRAN is to install the whole tidyverse.
+
+
+```r
+install.packages("tidyverse")
+```
+
+Alternatively, install just readr from CRAN:
+
+
+```r
+install.packages("readr")
+```
+
+readr is part of the core tidyverse, so load it with:
+
+
+```{r}
+library(tidyverse)
+```
+
+## Breaking Changes
+
+### Argument name consistency
+
+The first argument to all of the `write_()` functions, like `write_csv()` had previously been `path`.
+However the first argument to all of the `read_()` functions is `file`.
+As of readr 1.4.0 the first argument to both `read_()` and `write_()` functions is `file` and `path` is now deprecated.
+
+### NaN behavior
+
+Some floating point operations can produce a `NaN` value, e.g. `0 / 0`.
+Previously `write_csv()` would output `NaN` values always as `NaN` and this could not be controlled by the `write_csv(na=)` argument.
+Now the output value of `NaN` is the same as the `NA` and can be controlled by the argument.
+This is a breaking change in that the same code would produce different output, but it should be rare in practice.
+
+## New features
+
+### Generate column specifications from datasets
+
+Using `as.col_spec()` on any `data.frame` or `tibble` object will now generate a column specification with the column types in the data.
```

Comment:
This is cool!

Reply:
(It probably should've been `as_col_spec()` though)

Reply:
It is not a new generic, it has always been `as.col_spec()` since it was [added in 2015]( (https://github.com/tidyverse/readr/commit/ac24e9ed43b110b0666982af613971f361731c43). In this release I just added a method for data.frame objects.